src/WarcraftLegacies.Source/Quests/Scarlet/QuestReconquerLordaeron.cs

```diff
@@ -0,0 +1,56 @@
+﻿using MacroTools.ControlPointSystem;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.ObjectiveSystem.Objectives.ControlPointBased;
+using MacroTools.ObjectiveSystem.Objectives.FactionBased;
+using MacroTools.ObjectiveSystem.Objectives.QuestBased;
+using MacroTools.ObjectiveSystem.Objectives.UnitBased;
+using MacroTools.QuestSystem;
+using System.Collections.Generic;
+using MacroTools.LegendSystem;
+using WarcraftLegacies.Source.Objectives;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+using MacroTools.ObjectiveSystem.Objectives.TimeBased;
+using MacroTools.ObjectiveSystem.Objectives.LegendBased;
+
+namespace WarcraftLegacies.Source.Quests.Scarlet
+{
+  /// <summary>
+  /// Rebuild all the towns in Lordaeron to unlock the Divine Bastion and it's units.
+  /// </summary>
+  public sealed class QuestReconquerLordaeron : QuestData
+  {
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QuestReconquerLordaeron"/> class.
+    /// </summary>
+    public QuestReconquerLordaeron(QuestData stratholme, QuestData capital, QuestData hearthglen, QuestData brill, QuestData andorhal) : base(
+      "Reconquer Lordaeron",
+      "The Scarlet Crusade ultimate objective is to purge all undead from the Plaguelands and rebuild Lordaeron to it's glory.",
+      @"ReplaceableTextures/CommandButtons/BTNSalvationSpire.blp")
+    {
+      Required = true;
+      AddObjective(new ObjectiveCompleteQuest(stratholme));
+      AddObjective(new ObjectiveCompleteQuest(capital));
+      AddObjective(new ObjectiveCompleteQuest(hearthglen));
+      AddObjective(new ObjectiveCompleteQuest(brill));
+      AddObjective(new ObjectiveCompleteQuest(andorhal));
+      ResearchId = Constants.UPGRADE_R09A_QUEST_COMPLETED_THE_DIMENSIONAL_SHIP; //TODO create research according to new title picked by writting team
+    }
+
+    /// <inheritdoc/>
+    protected override void OnComplete(Faction whichFaction)
+    {
+      
+    }
+
+    /// <inheritdoc/>
+    protected override string RewardFlavour =>
+      "With the safeguard of Lordaeron, the Scarlet Crusade gains the highest blessing of the Light";
+
+    /// <inheritdoc/>
+    protected override string RewardDescription =>
+      "You will be able to train the Divine Bastion";
```

Comment:
I assume Divine Bastion is a building.
```suggestion
      "Learn to build the Divine Bastion";
```